### PR TITLE
[14.0] [MIG] purchase_stock_picking_return_invoicing

### PR DESCRIPTION
**Pull request by mariadforgeflow**
Standard migration to v14 
@MiquelRForgeFlow

### CODE REVIEW

File: purchase_stock_picking_return_invoicing/readme/CONTRIBUTORS.rst [hunk 1]
```diff
@@ -0,0 +1,11 @@
+* Forgeflow <https://www.forgeflow.com>:
+
+  * Jordi Ballester Alomar <jordi.ballester@eficent.com>
```
Comment (MiquelRForgeFlow) [inline purchase_stock_picking_return_invoicing/readme/CONTRIBUTORS.rst]:
```suggestion
  * Jordi Ballester Alomar <jordi.ballester@forgeflow.com>
```

File: purchase_stock_picking_return_invoicing/tests/test_purchase_stock_picking_return_invoicing.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
+# Copyright 2019 Eficent Business and IT Consulting Services
```
Comment (MiquelRForgeFlow) [inline purchase_stock_picking_return_invoicing/tests/test_purchase_stock_picking_return_invoicing.py]: ...

File: purchase_stock_picking_return_invoicing/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# Copyright 2017-2018 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Purchase Stock Picking Return Invoicing",
+    "summary": "Add an option to refund returned pickings",
+    "version": "14.0.1.0.0",
+    "category": "Purchases",
+    "website": "https://github.com/OCA/account-invoicing",
+    "author": "ForgeFlow," "Tecnativa," "Odoo Community Association (OCA)",
```
Comment (MiquelRForgeFlow) [inline purchase_stock_picking_return_invoicing/__manifest__.py]:
```suggestion
    "author": "ForgeFlow, Tecnativa, Odoo Community Association (OCA)",
```

File: purchase_stock_picking_return_invoicing/tests/test_purchase_stock_picking_return_invoicing.py [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+# Copyright 2017-2018 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields
+from odoo.tests.common import Form, SavepointCase
+
+
+class TestPurchaseStockPickingReturnInvoicing(SavepointCase):
+    at_install = False
+    post_install = True
```
Comment (MiquelRForgeFlow) [inline purchase_stock_picking_return_invoicing/tests/test_purchase_stock_picking_return_invoicing.py]:
```suggestion
from odoo.tests.common import Form, SavepointCase, tagged

@tagged('post_install', '-at_install')
class TestPurchaseStockPickingReturnInvoicing(SavepointCase):
```

File: purchase_stock_picking_return_invoicing/views/purchase_view.xml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+                    name="action_create_invoice_refund"
+                    string="Create Refund"
+                    type="object"
+                    class="oe_stat_button"
+                    context="{'create_refund':True}"
+                    attrs="{'invisible': ['|', ('state', 'not in', ('purchase', 'done')), ('invoice_status', 'in', ('no', 'invoiced'))]}"
+                />
+            </button>
+            <xpath
+                expr="//div[hasclass('oe_button_box')]/button[@name='action_create_invoice']"
```
Comment (MiquelRForgeFlow) [inline purchase_stock_picking_return_invoicing/views/purchase_view.xml]:
```suggestion
                expr="//div[hasclass('oe_button_box')]/button[@name='action_view_invoice']"
```

File: purchase_stock_picking_return_invoicing/views/purchase_view.xml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+                position="after"
+            >
+                <button
+                    type="object"
+                    name="action_create_invoice_refund"
```
Comment (MiquelRForgeFlow) [inline purchase_stock_picking_return_invoicing/views/purchase_view.xml]:
```suggestion
                    name="action_view_invoice_refund"
```

**Review by DavidBForgeFlow — APPROVED** [review]
Tested on runbot. LGTM.

**Review by RaulOForgeFlow — APPROVED** [review]
Functional review :+1:

**Review by AaronHForgeFlow — COMMENTED** [review]
Functional review LGTM. Just wondering because there is this symbol in the view

:

**Comment by MiquelRForgeFlow** [comment]
In v11:

Selection_201


Who knows, it was put by @pedrobaeza in https://github.com/OCA/account-invoicing/commit/54813327b0ce6fa439bffa35ed82333d4d1ca997#diff-c01a80d339d66395b2b2362492df10cc443e15832ca25a07cd1dd177efbfb6deR110

**Review by AaronHForgeFlow — APPROVED** [review]
Ok, that is a minor thing, then :+1: .

**Comment by pedrobaeza** [comment]
The asterisk is there because it's not true that it's the returned qty. It's the returned qty that you have marked to be invoiced. The help field also explains it.

**Review by andreampiovesana — APPROVED** [review]
merge!

**Review by MiquelRForgeFlow — APPROVED** [review]

**Pull request merged**